controllers/admin/AdminImportController.php

```diff
@@ -2104,17 +2104,14 @@ protected function productImportOne($info, $default_language_id, $id_lang, $forc
                 $id_product_supplier = (int) ProductSupplier::getIdByProductAndSupplier((int) $product->id, 0, (int) $product->id_supplier);
                 if ($id_product_supplier) {
                     $product_supplier = new ProductSupplier($id_product_supplier);
-                } else {
-                    $product_supplier = new ProductSupplier();
+                    $product_supplier->id_product = (int) $product->id;
```

Comment:
I'm not sure about this modification, it means that importing a new supplier that doesn't exist will never work anymore so in most cases import of product supplier will remain empty..

Reply:
@jolelievre that's actually what happens when you create a product with product page v1 when you don't specify a supplier, that's why we made that choice.

Reply:
Hmm but I'm not sure it is the same use case 🤔 In product page we are editing a product which can have an existing ProductSupplier, in this case we are importing a product, so most of the time a new product so the chances are high that the ProductSupplier will never exist for this couple of `id_product/id_supplier` meaning if it's not present the association is never created

I have close to zero knowledge about the import feature, but it really feels like in this case the ProductSupplier can only be updated when it exists and can never created be created by import

Reply:
@jolelievre I'm not sure to understand why it's different?
In product page you can also create a product without defining a product supplier so to me it's like importing without specifying a product supplier?
Also, if I remember well, you can have a product supplier during import, you just have to specify which one

Reply:
I think the main thing to check here is if `!empty($product->id_supplier)` to make sure we don't create an association to a non existing supplier

But if the association doesn't exist (and that's the purpose of `ProductSupplier::getIdByProductAndSupplier`) we still need to create it (and not update it)

Reply:
@jolelievre you're right, I didn't get it right 😅

Reply:
I am a bit lost with this issue so I may be out of the subject.
But @jolelievre , isn't the test you are asking for done at line 2103 ?

```php
if (!$validateOnly && !empty($product->id) && property_exists($product, 'supplier_reference')) {
```

Reply:
Hmm, I'm not talking about the reference value but about the actual `$product->id_supplier` which stores the relation between the product and its default supplier

What I mean in this comment is:
- with your modification, you only create the `ProductSupplier` when `$id_product_supplier` is not empty
- `$id_product_supplier` is initialized with `ProductSupplier::getIdByProductAndSupplier` which searches for a relationship between the product and `$product->id_supplier`, meaning it will only return a value if the supplier is already associated to the product (ProductSupplier represents the association table between product and supplier)
- so if the relationship isn't found (because it's not in DB) you don't create the ProductSupplier, meaning you don't create the relationship
- so an inexistent relationship will never be created
- so with you modify you can only modify/update an existing relationship
- TLDR: if the relationship doesn't exist yet it's never created so the import is now unable to create a new relationship (it was able to do so with the previous code)

So that's why I'm blocking this modification because it introduces a regression

Reply:
Now besides this modification which is not appropriate I discussed with @atomiix who worked with you on this bug fix and it seems like there was a bug because of this because a ProductSupplier instance was created but the id_supplier was 0 which is not accepted in the product page v2 because it tries to get a non-existent supplier (indeed no supplier can have an ID 0) But the problem comes from the fact that we created a ProductSuplier relationship with id_supplier being 0

This should be prevented, and to do so we need to check that `$product->id_supplier` is different from 0 before we try and create the ProductSupplier (and we don't even need to look for a relationship with `ProductSupplier::getIdByProductAndSupplier` since it doesn't make sense)

Reply:
TLDR: you should add `&& !empty($product->id_supplier)` on Line 2103, currently the test checks the `supplier_reference` value not the ID itself